File: QRCodeTest/QRCodeTest/ViewController.m
//
//  ViewController.m
//  QRCodeTest
//
//  Created by daoquan on 16/9/12.
//  Copyright © 2016年 daoquan. All rights reserved.
//

#import "ViewController.h"
#import "QRCodeScanVC.h"

@interface ViewController ()

@property(nonatomic, strong)UIButton *scanbtn;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _scanbtn = [UIButton buttonWithType:UIButtonTypeCustom];
    [_scanbtn setTitle:@"扫描" forState:UIControlStateNormal];
    [_scanbtn setBackgroundColor:[UIColor orangeColor]];
    [_scanbtn addTarget:self action:@selector(startScan) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:_scanbtn];
    
    _scanbtn.frame = CGRectMake(0, 0, 80, 40);
    _scanbtn.center = CGPointMake(CGRectGetMidX([UIScreen mainScreen].bounds), CGRectGetMidY([UIScreen mainScreen].bounds));
    self.view.backgroundColor = [UIColor whiteColor];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark-> 二维码扫描
-(void)startScan{
    QRCodeScanVC *scanVC = [[QRCodeScanVC alloc]init];
//    Scan_VC*vc=[[Scan_VC alloc]init];
    [self.navigationController pushViewController:scanVC animated:YES];
}

@end

File: 11_SwiftLintExample/SwiftLintExample/ViewController.swift
//
//  ViewController.swift
//  SwiftLintExample
//
//  Created by daoquan on 2018/1/24.
//  Copyright © 2018年 deerdev. All rights reserved.
//  swiftlint:disable identifier_name

import UIKit

class ViewController: UIViewController {

    // swiftlint:disable colon
    var name:String = "deerdev"
    // swiftlint:enable colon

    var MyAge: Int = 18

    @objc  var nal = "sss"

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        UIView.animate(withDuration: 1) {
            print("\(self.name) is \(self.MyAge) years old")
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}
